### PR TITLE
Faster and more detailed analysis

### PR DESCRIPTION
**Pull request by tomato42**
### Description

Make box plot generation really fast, make other graphs optional
Add support for generating the plot of mean differences, together with confidence intervals

(depends on  #686)

### Motivation and Context

Make the analysis more useful for analysing very small differences 

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [x] the changes are also reflected in documentation and code comments
- [x] all new and existing tests pass (see Travis CI results)
- [ ] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [x] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [x] new and modified scripts were ran against popular TLS implementations:
  - [x] OpenSSL
  - [x] NSS
  - [x] GnuTLS
- [x] required version of tlslite-ng updated in requirements.txt and README.md

### TODO
 - [x] transpose the timing results in `timing.csv` so that reading them is much faster
 - [x] add a script that can combine multiple `timing.csv` into one single big one
   - [x] add documentation for above
 - [x] make all plots larger

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/689)

### CODE REVIEW

File: docs/source/timing-analysis.rst [hunk 1]
```diff
@@ -309,6 +315,37 @@ file:
...
+Combining results from multiple runs
+------------------------------------
+
+You can use the ``combine.py`` script to combine the results from runs.
+
+The script checks if the set of executed probes match in all the files,
+but you need to ensure that the environments of the test execution match
+too.
+
+To combine the runs provide the output directory (``out-dir`` here) and
```
Comment (sahanaprasad07) [inline docs/source/timing-analysis.rst:327]: Nit: To combine the runs,

File: docs/source/timing-analysis.rst [hunk 2]
```diff
@@ -309,6 +315,37 @@ file:
...
+.. code:: bash
+
+   PYTHONPATH=. python tlsfuzzer/combine.py -o out-dir \
+   in_1596892760/timing.csv in_1596892742/timing.csv
+
+.. warning::
+
+   The script overwrites the ``timing.csv`` in the output directory!
+
+After combining the the ``timing.csv`` files execute analysis as usual.
```
Comment (sahanaprasad07) [inline docs/source/timing-analysis.rst:339]: Nit: files,

File: docs/source/timing-analysis.rst (hunk 1 above)
Comment (tomato42) [reply docs/source/timing-analysis.rst]: fixed

File: docs/source/timing-analysis.rst (hunk 2 above)
Comment (tomato42) [reply docs/source/timing-analysis.rst]: fixed

**Pull request merged**